tests/next_tests/integration_tests/feature_tests/cases.py

```diff
@@ -0,0 +1,358 @@
+from typing import Any, Callable, Optional, Protocol, TypeAlias
+
+import numpy as np
+import pytest
+from typing_extensions import Self
+
+from gt4py.next import common
+from gt4py.next.ffront import decorator, fbuiltins
+from gt4py.next.iterator import embedded, ir as itir
+from gt4py.next.program_processors import processor_interface as ppi
+from gt4py.next.type_system import type_specifications as ts, type_translation
+
+from next_tests.integration_tests.feature_tests.ffront_tests.ffront_test_utils import (  # noqa: F401 # import alias for later moving this here; noqa: F401: # fixture
+    Edge,
+    IDim,
+    Ioff,
+    JDim,
+    Joff,
+    KDim,
+    Koff,
+    Vertex,
+    fieldview_backend,
+)
+
+
+IField: TypeAlias = common.Field[[IDim], int]
+IJKField: TypeAlias = common.Field[[IDim, JDim, KDim], int]
+IJKFloatField: TypeAlias = common.Field[[IDim, JDim, KDim], np.float64]
+VField: TypeAlias = common.Field[[Vertex], int]
+EField: TypeAlias = common.Field[[Edge], int]
+V2EDim = common.Dimension("V2E", kind=common.DimensionKind.LOCAL)
+E2VDim = common.Dimension("E2V", kind=common.DimensionKind.LOCAL)
+V2E = fbuiltins.FieldOffset("V2E", source=Edge, target=(Vertex, V2EDim))
+E2V = fbuiltins.FieldOffset("E2V", source=Vertex, target=(Edge, E2VDim))
+
+
+def no_backend(program: itir.FencilDefinition, *args, **kwargs) -> None:
+    """Temporary default backend to avoid accidentally testing the wrong backend."""
+    raise ValueError("No backend selected! Backend selection is mandatory in tests.")
+
+
+class ScalarInitializer(Protocol):
+    def __call__(self, dtype: str, shape: tuple[int, ...]) -> int | float | np.array:
+        ...
+
+
+class FieldInitializer(Protocol):
+    def __call__(
+        self, backend: ppi.ProgramProcessor, sizes: dict[common.Dimension, int], dtype: str
+    ) -> common.Field:
+        ...
+
+
+def zeros(
+    backend: ppi.ProgramProcessor, sizes: dict[common.Dimension, int], dtype: str
+) -> common.Field:
+    """Initialize a field with all zeros."""
+    return embedded.np_as_located_field(*sizes.keys())(np.zeros(tuple(sizes.values()), dtype=dtype))
+
+
+def unique(
+    backend: ppi.ProgramProcessor, sizes: dict[common.Dimension, int], dtype: str
+) -> common.Field:
+    """Initialize a field with a unique value in each coordinate."""
+    svals = tuple(sizes.values())
+    return embedded.np_as_located_field(*sizes.keys())(
+        np.arange(np.prod(svals), dtype=dtype).reshape(svals)
+    )
+
+
+def scalar5(dtype: str, shape: Optional[tuple[int, ...]]) -> int | float | np.array:
+    """Initialize a scalar to the value 5."""
+    if shape:
+        return np.ones(np.prod(shape), dtype=dtype) * 5
+    return np.dtype(dtype).type(5)
+
+
+@dataclasses.dataclass(frozen=True)
+class Builder:
+    partial: functools.partial
+
+    def build(self, *args, **kwargs) -> Self:
+        return self.partial(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs) -> Any:
+        return self.build(*args, **kwargs)
+
```

Comment:
I think a custom `__getattr__()` should be added here  (maybe raising a custom message error) to avoid complaints from the static type checker, since the attribute setters are added at runtime.